### ISSUE TITLE
Doesn't validate realm on return

### ISSUE THREAD

**Issue by switz** [liamcurry/passport-steam#24]
This library (or one of its dependencies) does not validate the openid realm. If I login to another third party site via steam, they can log in to my site using the secret.

Let me know if this is upstream and I'll gladly pass it along. If it is upon me to validate the realm, it would be very helpful if you can provide an example.

Thanks!

**Comment by welps** [comment]
I suspect it's the passport-openid dependency that should be validating this, but I'll take this weekend to confirm and fix, if possible. Thanks.

**Comment by welps** [comment]
@switz I'm having issues reproducing this, do you have a proof of concept / example I can check out?

**Comment by welps** [comment]
Not sure if this helps you in the short term but you can check the request itself when verifying the user if you set `passReqToCallback` to true when passing in `SteamStrategy`.

````
passport.use(new SteamStrategy({
        returnURL: 'returnURL',
        realm: 'realmURL',
        apiKey: 'yourAPIkey',
        passReqToCallback: true
    },
    function(req, identifier, profile, done) {
        process.nextTick(function () {
            profile.identifier = identifier;
            return done(null, profile);
        });
    }
));
````

**Comment by switz** [comment]
Here's the fix: https://github.com/havard/node-openid/pull/141

**Comment by welps** [comment]
Oh. That package is a dependency of passport-openid so passport-steam will pick it up once it gets merged and updated on npm. 

Thanks for reporting this. I'll keep track of the issue progress since that's quite the security hole.

**Comment by havard** [comment]
Fixed in 2.0.0, note the major version bump

**Comment by welps** [comment]
Thanks for the resolution and update, really appreciate it!
